### PR TITLE
docs: Add information on how a SG account is linked to an auth provider account

### PR DESCRIPTION
**Pull request by miveronese**
Updating our docs to make sure we are more clear when informing our customers how an SG account is linked to an auth provider account.
This is based on a recent question raised by a customer that thought it would be possible to link their SG account to an auth provider via username only.

## Test plan
Manually tested, in my local instance, that the links are working and the updated page is showing the changes.
To better read/review the changes, you can [use this link](https://github.com/sourcegraph/sourcegraph/blob/a316dfe3fd737fbf61843b009aa2b112b3192514/doc/admin/auth/index.md#linking-a-sourcegraph-account-to-an-auth-provider).

### CODE REVIEW

**Review by jplahn — APPROVED** [review]

**Review by ryphil — APPROVED** [review]

File: doc/admin/auth/index.md [hunk 1]
```diff
@@ -437,6 +437,14 @@ Some proxies add a prefix to the username header value. For example, Google IAP
   ]
 }
 ```
+
+## Linking a Sourcegraph account to an auth provider
+
+The link between a Sourcegraph account and an authentication provider account happens via email only.
```
Comment (unknwon) [inline doc/admin/auth/index.md:443]: IIRC Bitbucket Server and HTTP proxy header are through usernames

Comment (unknwon) [reply doc/admin/auth/index.md:443]: In case you missed it 👆 @miveronese

Comment (miveronese) [reply doc/admin/auth/index.md:443]: I missed that! Was reviewing old PRs when I saw this! will push a fix now. Thanks for the heads up, @unknwon

Comment (unknwon) [reply doc/admin/auth/index.md:443]: Thank you!

**Pull request merged**